### PR TITLE
add file snapshot assertion

### PR DESCRIPTION
**Pull request by SjorsO**
As proposed in #34

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -159,6 +160,13 @@ As a result, our snapshot file returns "bar" instead of "foo".
 <?php return 'bar';
 ```
 
+### File snapshots
```
Comment (sebastiandedeyne) [inline README.md]: Needs a line of whitespace after

File: src/Filesystem.php [hunk 2]
```diff
@@ -27,6 +27,35 @@ public function has(string $filename): bool
         return file_exists($this->path($filename));
     }
 
+    /**
+     * Get all file names in this directory that have the same name
+     * as $fileName, but have a different file extension.
+     * @param string $fileName
```
Comment (sebastiandedeyne) [inline src/Filesystem.php]: Add an empty line after the description

File: src/Filesystem.php [hunk 3]
```diff
@@ -27,6 +27,35 @@ public function has(string $filename): bool
...
+     * @return array
```
Comment (sebastiandedeyne) [inline src/Filesystem.php]: Add an empty line after the params

**Comment by sebastiandedeyne** [comment]
Hi @SjorsO 

Requested a few whitespace nitpicks.

I like the addition of the method, but I'm still pretty torn about saving the files that failed...

**Comment by sebastiandedeyne** [comment]
Gonna merge this once you updated according to the mini-review :)

**Comment by SjorsO** [comment]
Hey @sebastiandedeyne 

I've committed the formatting changes.

And about saving failed files: we could also make saving failed files optional by making the user override a `protected function shouldSaveFailedSnapshotFiles(): bool` function. I've noticed that especially during development it is really useful to be able to manually compare why the test failed, so i'd like to keep this functionality (but off by default is fine by me)

**Comment by sebastiandedeyne** [comment]
Gonna leave it as is, thanks! 🙌

**Pull request merged**